File: API/Entities/Product.cs
using API.Entities.Enums;

namespace API.Entities;

public class Product
{
    public int Id { get; set; }

    public required string Name { get; set; }
    public string NormalizedName { get; set; }

    public string Description { get; set; }

    public int CategoryId { get; set; }
    public ProductCategory Category { get; set; }
    public Dictionary<int, float> Prices { get; set; }
    /// <summary>
    /// This value is valid inside a category, not between
    /// </summary>
    public int SortValue { get; set; }
    public ProductType  Type { get; set; }

    public bool IsTracked { get; set; }
    public bool Enabled { get; set; }
}

File: UI/Web/src/app/_models/product.ts
export type Product = {
  id: number;
  name: string;
  description: string;
  categoryId: number;
  sortValue: number;
  type: ProductType;
  isTracked: boolean;
  enabled: boolean;
  prices: Record<string, number>;
};

export enum ProductType {
  Consumable = 0,
  OneTime = 1,
}

export const AllProductTypes = [ProductType.Consumable, ProductType.OneTime];

export type ProductCategory = {
  id: number;
  name: string;
  enabled: boolean;
  autoCollapse: boolean;
  sortValue: number;
}

export type PriceCategory = {
  id: number;
  name: string;
}
